docs/fiddles/ipc/pattern-1/main.js

```diff
@@ -1,28 +1,24 @@
 const {app, BrowserWindow, ipcMain} = require('electron')
 const path = require('path')
 
+function handleSetTitle (event, title) {
+  const webContents = event.sender
+  const win = BrowserWindow.fromWebContents(webContents)
+  win.setTitle(title)
+}
+
 function createWindow () {
   const mainWindow = new BrowserWindow({
     webPreferences: {
       preload: path.join(__dirname, 'preload.js')
     }
   })
-
-  ipcMain.on('set-title', (event, title) => {
-    const webContents = event.sender
-    const win = BrowserWindow.fromWebContents(webContents)
-    win.setTitle(title)
-  })
-
   mainWindow.loadFile('index.html')
 }
 
 app.whenReady().then(() => {
```

Comment:
The handleChangeTitle function is definitely a better way to handle the 'set-title' message, well done.